src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.cs

```diff
@@ -556,4 +556,61 @@ public async Task SupportsHandlersWithSameSignatureButDifferentParameterNamesFro
         Assert.Equal(new EventId(5, "ImplicitBodyNotProvided"), log2.EventId);
         Assert.Equal(@"Implicit body inferred for parameter ""todo1"" but no body was provided. Did you mean to use a Service instead?", log2.Message);
     }
+
+    [Fact]
+    public async Task SkipsMapGetWithIncorrectNamespaceAndAssembly()
+    {
+        var source = """
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+
+public static class TestMapActions
+{
+    public static IEndpointRouteBuilder MapTestEndpoints(this IEndpointRouteBuilder app)
+    {
+        app.ServiceProvider.Map(1, (string test) => "Hello world!");
+        app.ServiceProvider.MapPost(2, (string test) => "Hello world!");
+        return app;
+    }
+}
+
+public static class EndpointRouteBuilderExtensions
+{
+    public static IServiceProvider Map(this IServiceProvider app, int id, Delegate requestDelegate)
```

Comment:
Another nit: Should use the exact same signature as Map
```
public static IEndpointConventionBuilder Map(
        this IEndpointRouteBuilder endpoints,
        RoutePattern pattern,
        RequestDelegate requestDelegate)
```

Reply:
Based on the overload resolution rules for extension methods, the compiler will bind to the methods defined in the framework if they have the same signature. I did update the tests to include a scenario with extension methods targeting `IEndpointRouteBuilder` and a different type for the second parameter.